docs/map-api-research.md

```diff
@@ -0,0 +1,115 @@
+| **Vector Tiles API**      | Retrieve vector-tile data                                                         | 200 000 requests            |
+| **Search (Geocoding)**    | Address & POI lookup by name or coords                                            | 100 000 requests            |
+| **Search (POI Category)** | List POIs by category (e.g. “coffee shops” around coords)                         | 50 000 requests             |
+| **Temporary Geocoding**   | Geocoding/place search without persistent indexing                                | 100 000 requests            |
+
+\*MAU = Monthly Active Users
+
+***
+
+## 2. Here Maps
+
+| API/SDK                         | Description                                                                            | Free Tier (Monthly)    |
+|---------------------------------|----------------------------------------------------------------------------------------|------------------------|
+| **Vector Tile**                 | Pre-rendered tiles & client-side vector rendering                                       | 30 000 requests        |
+| **Geocode / Reverse Geocode**   | Address ↔ geographic coordinate lookup                                                 | 30 000 requests        |
+| **Discover / Search**           | Category & keyword search for places                                                    | 5 000 requests         |
+| **Matrix Routing**              | Time/distance matrices between origins & destinations                                   | 2 500 requests         |
+| **Routing (Car/Bike/Walk)**     | Directions for multiple modes                                                          | 30 000 requests        |
+
+***
+
+## 3. Google Maps Platform
+
+| API                              | Description                                         | Free Tier (Monthly) |
+| -------------------------------- | --------------------------------------------------- | ------------------- |
+| **Maps JavaScript API**          | Interactive, styled maps                            | 10 000 loads        |
+| **Maps Embed API**               | Simple `<iframe>` maps & Street View                | Unlimited           |
+| **Routes: Compute Route Matrix** | Travel-time/distance matrix                         | 10 000 elements     |
+| **Routes: Compute Routes**       | Real-time directions (traffic, transit, bike, walk) | 10 000 requests     |
+| **Geocoding API**                | Forward & reverse geocoding                         | 10 000 requests     |
+| **Geolocation API**              | Device location via cell/WiFi data                  | 10 000 requests     |
+| **Places API — Place Details**   | Detailed info (addresses, opening hours, etc.)      | 10 000 requests     |
+|                                  |                                                     |                     |
```

Comment:
Add a note about Google’s billing requirements, e.g., "A credit card is required to activate the free tier, even for limited use."